source/src/microapp/cs_MicroappController.cpp

```diff
@@ -581,3 +593,14 @@ void MicroappController::incrementEmptySoftInterruptSlots() {
 	}
 	_emptySoftInterruptSlots++;
 }
+
+void MicroappController::clear(uint8_t appIndex) {
+	LOGi("Clear appIndex=%u", appIndex);
+	for (int i = 0; i < MICROAPP_MAX_SOFT_INTERRUPT_REGISTRATIONS; ++i) {
+		_softInterruptRegistrations[i] = {};
+	}
+	_emptySoftInterruptSlots = 1;
```

Comment:
Not sure, i just set it to the same value as it is initially.

Reply:
Naming is a bit confusing to be sure, but `_emptySoftInterruptSlots` refers to the number of slots on the microapp side for handling new interrupts. So it has nothing to do with `MICROAPP_MAX_SOFT_INTERRUPT_REGISTRATIONS`. Since the maximum for that is only defined on the microapp side, bluenet initially assumes a minimum number (1) and on yield requests, the microapp will update bluenet with the actual number and bluenet will then use that.